pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -0,0 +1,557 @@
+  }
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  health ` + nodeLocal + `:8080
+  }
+in-addr.arpa:53 {
+  errors
+  cache 30
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  }
+ip6.arpa:53 {
+  errors
+  cache 30
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  }
+.:53 {
+  errors
+  cache 30
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . __PILLAR__UPSTREAM__SERVERS__ {
+          ` + c.forceTcpToUpstreamDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  }
+`,
+			},
+		}
+	)
+	utilruntime.Must(kutil.MakeUnique(configMap))
+
+	var (
+		service = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      serviceName,
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					"k8s-app":                       "kube-dns-upstream",
+					"kubernetes.io/cluster-service": "true",
+					"kubernetes.io/name":            "NodeLocalDNS",
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Selector: map[string]string{"k8s-app": "kube-dns"},
+				Ports: []corev1.ServicePort{
+					{
+						Name:       "dns",
+						Port:       int32(PortServiceServer),
+						TargetPort: intstr.FromInt(PortServer),
+						Protocol:   corev1.ProtocolUDP,
+					},
+					{
+						Name:       "dns-tcp",
+						Port:       int32(PortServiceServer),
+						TargetPort: intstr.FromInt(PortServer),
+						Protocol:   corev1.ProtocolTCP,
+					},
+				},
+			},
+		}
+
+		daemonset = &appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					LabelKey:                        LabelValue,
+					v1beta1constants.GardenRole:     v1beta1constants.GardenRoleSystemComponent,
+					managedresources.LabelKeyOrigin: managedresources.LabelValueGardener,
+				},
+			},
+			Spec: appsv1.DaemonSetSpec{
+				UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+					RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+						MaxUnavailable: &intstr.IntOrString{IntVal: 10},
+					},
+				},
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						LabelKey: LabelValue,
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							LabelKey:                                 LabelValue,
+							v1beta1constants.LabelNetworkPolicyToDNS: "allowed",
+						},
+						Annotations: map[string]string{
+							"prometheus.io/port":   strconv.Itoa(prometheusPort),
+							"prometheus.io/scrape": strconv.FormatBool(prometheusScrape),
+						},
```

Comment:
See https://github.com/gardener/gardener/pull/5099/files#diff-e51d1ba4023287933a40a6da60524548e4ed77eda1052a8b9b8d1a85a0695ddeR32-R33, can you add this annotation as well?